src/DataCollection.php

```diff
@@ -233,21 +255,42 @@ public function fetchDocument($documentId, array $options = [])
      * Retrieves all documents stored in this data collection.
      *
      * @param array $options Optional parameters
-     * @return AdvancedSearchResult containing the total number of retrieved documents and an array of Kuzzle/Document objects
+     * @return Document[] containing all documents objects
      */
     public function fetchAllDocuments(array $options = [])
     {
+        $documents = [];
         $filters = [];
 
+        if (array_key_exists('scroll', $options)) {
+            $filters['scroll'] = $options['scroll'];
+            unset($options['scroll']);
+        }
+
         if (array_key_exists('from', $options)) {
             $filters['from'] = $options['from'];
+            unset($options['from']);
+        } else {
+            $filters['from'] = 0;
         }
 
         if (array_key_exists('size', $options)) {
             $filters['size'] = $options['size'];
+            unset($options['size']);
+        } else {
+            $filters['size'] = 1000;
+        }
+
+        $searchResult = $this->search($filters, $options);
+
+        while ($searchResult) {
```

Comment:
I'm not confortable with this solution: if there are tens of thousands documents (or millions of them), you'll saturate the memory with no mean for the client to know what's going on.

I would either return a paginated result, like `search` does, or put a hard limit on the maximum number of documents that can be fetched that way, with an explicit message advising the use of `search` when the number of documents is too high.